File: data/derived_cot/rq3_prompting/results/translation/r1/java_to_cpp/code_output/BalancedBrackets.cpp
#include <iostream>
#include <stack>
#include <string>

using namespace std;

class BalancedBrackets {
private:
    stack<char> stack;
    string leftBrackets;
    string rightBrackets;
    string expr;
public:
    BalancedBrackets(const string& expr) 
        : leftBrackets("({["), rightBrackets(")}]"), expr(expr) {}

    void clearExpr() {
        string cleanedExpr;
        for (char c : expr) {
            if (leftBrackets.find(c) != string::npos || 
                rightBrackets.find(c) != string::npos) {
                cleanedExpr += c;
            }
        }
        expr = cleanedExpr;
    }

    bool checkBalancedBrackets() {
        clearExpr();
        for (char Brkt : expr) {
            if (leftBrackets.find(Brkt) != string::npos) {
                stack.push(Brkt);
            } else {
                if (stack.empty()) {
                    return false;
                }
                char currentBrkt = stack.top();
                stack.pop();
                if (currentBrkt == '(' && Brkt != ')') return false;
                if (currentBrkt == '{' && Brkt != '}') return false;
                if (currentBrkt == '[' && Brkt != ']') return false;
            }
        }
        return stack.empty();
    }

    const string& getExpr() const {
        return expr;
    }
};

int main() {
    BalancedBrackets b("a(b)c");
    cout << boolalpha << b.checkBalancedBrackets() << endl;
    return 0;
}

File: data/derived_cot/rq1_traces/translation/glm5.1/cpp_to_py/code_output/BalancedBrackets.py
class BalancedBrackets:
    leftBrackets = "({["
    rightBrackets = ")}]"

    def __init__(self, expr: str):
        self.stack = []
        self.expr = expr

    def clearExpr(self):
        self.expr = "".join(
            c for c in self.expr 
            if c in self.leftBrackets or c in self.rightBrackets
        )

    def check_balanced_brackets(self):
        self.clearExpr()
        for Brkt in self.expr:
            if Brkt in self.leftBrackets:
                self.stack.append(Brkt)
            else:
                if not self.stack:
                    return False
                Current_Brkt = self.stack.pop()
                if Current_Brkt == '(' and Brkt != ')':
                    return False
                if Current_Brkt == '{' and Brkt != '}':
                    return False
                if Current_Brkt == '[' and Brkt != ']':
                    return False
        return not self.stack